cli/command/environment_test.go

```diff
@@ -0,0 +1,312 @@
+		{
+			name:    "update",
+			command: command.update,
+		},
+		{
+			name:    "setMinCount",
+			command: command.update,
+			args:    []string{"env_name", "1w"},
+		},
+		{
+			name:    "read",
+			command: command.read,
+		},
+		{
+			name:    "delete",
+			command: command.delete,
+		},
+		{
+			name:    "link",
+			command: command.link,
+		},
+		{
+			name:    "unlink",
+			command: command.unlink,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := getCLIContext(t, tc.args, nil)
+
+			if err := tc.command(c); err == nil {
+				t.Fatalf("%s: error was nil!", tc.name)
+			}
+		})
+	}
+}
+
+func TestCreateEnvironment(t *testing.T) {
+	testNoWaitCaseHelper(t, func(t *testing.T, otherFlags map[string]interface{}, wait bool) {
+		base, ctrl := newTestCommand(t)
+		defer ctrl.Finish()
+
+		command := NewEnvironmentCommand(base.Command())
+
+		userData := "user_data"
+		file, close := tempFile(t, userData)
+		defer close()
+
+		req := models.CreateEnvironmentRequest{
+			EnvironmentName:  "env_name",
+			InstanceSize:     "m3.large",
+			MinClusterCount:  2,
+			UserDataTemplate: []byte(userData),
+			OperatingSystem:  "linux",
+			AMIID:            "ami",
+		}
+
+		environment := &models.Environment{}
+		job := &models.Job{
+			JobID:  "job_id",
+			Status: job.Completed.String(),
+			Result: "entity_id",
+		}
+
+		base.Client.EXPECT().
+			CreateEnvironment(req).
+			Return(job.JobID, nil)
+
+		if wait {
+			base.Client.EXPECT().
+				ReadJob(job.JobID).
+				Return(job, nil)
+
+			base.Client.EXPECT().
+				ReadEnvironment(job.Result).
+				Return(environment, nil)
+		}
+
+		flags := map[string]interface{}{
+			"size":      req.InstanceSize,
+			"min-count": req.MinClusterCount,
+			"user-data": file.Name(),
+			"os":        req.OperatingSystem,
+			"ami":       req.AMIID,
+		}
+		for k, v := range otherFlags {
+			flags[k] = v
+		}
+
+		c := getCLIContext(t, []string{"env_name"}, flags)
+		if err := command.create(c); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestDeleteEnvironment(t *testing.T) {
+	testNoWaitCaseHelper(t, func(t *testing.T, flags map[string]interface{}, wait bool) {
+		base, ctrl := newTestCommand(t)
+		defer ctrl.Finish()
+
+		command := NewEnvironmentCommand(base.Command())
+
+		job := &models.Job{
+			JobID:  "job_id",
+			Status: job.Completed.String(),
+			Result: "entity_id",
+		}
+
+		base.Resolver.EXPECT().
+			Resolve("environment", "env_name").
+			Return([]string{"env_id"}, nil)
+
+		base.Client.EXPECT().
+			DeleteEnvironment("env_id").
+			Return(job.JobID, nil)
+
+		if wait {
+			base.Client.EXPECT().
+				ReadJob(job.JobID).
+				Return(job, nil)
+		}
+
+		c := getCLIContext(t, []string{"env_name"}, flags)
+		if err := command.delete(c); err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func TestGetEnvironment(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	command := NewEnvironmentCommand(base.Command())
+
+	base.Resolver.EXPECT().
```

Comment:
We should probably adjust this to ensure we call ReadEnvironment on multiple resolved ids

Reply:
✅